py3status/modules/hueshift.py

```diff
@@ -0,0 +1,185 @@
+    button_down: mouse button to change color temperature (default 5)
+    button_reset: mouse button to reset color temperature (default 3)
+    button_up: mouse button to change color temperature (default 4)
+    command: specify blueshift, redshift, or sct to use, otherwise automatic
+        (default None)
+    delta: specify interval value to change color temperature (default 250)
+    format: display format for this module
+        *(default '\?if=pgrep&color=bad \u263c!'
+        '|\?color=hue \u263c {temperature}K')*
+    initial: specify initial color temperature to use (default 6500)
+    maximum: specify maximum color temperature to use (default 25000)
+    minimum: specify minimum color temperature to use (default 1000)
+    thresholds: specify color thresholds to use
+        (default [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')])
+
+Control placeholder:
+    {pgrep} a boolean based on pgrep processing data
+
+Format placeholder:
+    {temperature} color temperature, eg 6500
+    {command} command name, eg blueshift, redshift, sct
+
+Color options:
+    color_bad: disabled due to running processes
+
+Color thresholds:
+    hue: print a color based on the value of color temperature
+
+Requires:
+    blueshift: an extensible and highly configurable alternative
+    redshift: program to adjust the color temperature of your screen
+    sct: set color temperature
+
+Suggestions:
+    campfire: 4500
+    dust storm on mars: 2000
+    coffee free all nighter: 8000
+
+Examples:
+```
+# good rules; show command name; don't colorize everything
+hueshift {
+    format = '[\?if=pgrep&color=bad \u263c!|[\?color=hue&show \u263c '
+    format += '{command}] {temperature}K]'
+    minimum = 3000
+    maximum = 10000
+}
+```
+
+@author lasers
+
+SAMPLE OUTPUT
+# warm
+{'full_text': u'\u263c 3000K', 'color': #f3c},
+
+# neutral
+{'full_text': u'\u263c 6500K', 'color': #ff3}
+
+# cool
+{'full_text': u'\u263c 10000K', 'color': #3cf},
+
+paused
+{'full_text': u'\u263c!', 'color': #f00}
+"""
+
+STRING_NOT_INSTALLED = 'not installed'
+STRING_BAD_COMMAND = 'invalid command'
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_down = 5
+    button_reset = 3
+    button_up = 4
+    command = None
+    delta = 250
+    format = '\?if=pgrep&color=bad \u263c!|\?color=hue \u263c {temperature}K'
+    initial = 6500
+    maximum = 25000
+    minimum = 1000
+    thresholds = [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')]
+
+    def post_config_hook(self):
+        hueshift_commands = ['blueshift', 'redshift', 'sct']
+        self.pgrep_command = ['pgrep', '|'.join(hueshift_commands)]
+        if not self.command:
+            self.command = self.py3.check_commands(hueshift_commands)
+        elif self.command not in hueshift_commands:
+            raise Exception(STRING_BAD_COMMAND)
+        elif not self.py3.check_commands(self.command):
+            raise Exception('%s %s' % (self.command, STRING_NOT_INSTALLED))
```

Comment:
this is bad for translation (yes I know we have none but I think it is good practice)
```
STRING_NOT_INSTALLED = '{} not installed'
raise Exception(STRING_NOT_INSTALLED.format(self.command))
```
is nicer :)

Reply:
Keep this as-is... Similar to `volume_status`.
* `hueshift: redshift not installed`   # if command is redshift
* `hueshift: blueshift not installed`   # if command is blueshift
* `hueshift: redshift not installed`   # if command is sct
* `hueshift not installed`   # if command is None

Reply:
The idea is you become a better coder, and py3status gets better not just keep doing the same old shit

Reply:
D'oh. I forget we changed the strings in `volume_status`. I'll do this too.